sql/core/src/test/scala/org/apache/spark/sql/DataFrameHintSuite.scala

```diff
@@ -42,43 +44,53 @@ class DataFrameHintSuite extends AnalysisTest with SharedSparkSession {
 
     check(
       df.hint("hint1", 1, "a"),
-      UnresolvedHint("hint1", Seq(1, "a"), df.logicalPlan)
+      UnresolvedHint("hint1", Seq(Literal(1), Literal("a")), df.logicalPlan)
     )
 
     check(
       df.hint("hint1", 1, $"a"),
-      UnresolvedHint("hint1", Seq(1, $"a"),
+      UnresolvedHint("hint1", Seq(Literal(1), $"a".expr),
         df.logicalPlan
       )
     )
 
     check(
-      df.hint("hint1", Seq(1, 2, 3), Seq($"a", $"b", $"c")),
-      UnresolvedHint("hint1", Seq(Seq(1, 2, 3), Seq($"a", $"b", $"c")),
+      df.hint("hint1", Array(1, 2, 3), array($"a", $"b", $"c")),
```

Comment:
is this a breaking change? so `Seq(1, 2, 3)` doesn't work in `df.hint` anymore?

Reply:
It's better to avoid breaking change unless it needs a lot of effort.

Is it only for `Seq[Int]`? Maybe we can special-case it.

Reply:
Oh I missed it. It's a custom hint `hint1`. I think we are fine as long as the builtin hints are not broken.